common/metrics/otel_metrics_handler.go

```diff
@@ -35,40 +36,51 @@ import (
 	"go.temporal.io/server/common/log/tag"
 )
 
-// MetricsHandler is an event.Handler for OpenTelemetry metrics.
-// Its Event method handles Metric events and ignores all others.
-type otelMetricsHandler struct {
-	l           log.Logger
-	tags        []Tag
-	provider    OpenTelemetryProvider
-	excludeTags excludeTags
-}
+type (
+	// otelMetricsHandler is a Handler for OpenTelemetry metrics.
+	otelMetricsHandler struct {
```

Comment:
How is this working for tally? I only see changes to otel handler.

Reply:
This PR only adds it for OTel. IIRC, we only needed it there. However, I can make it work for tally in a follow-up, too.